### PR TITLE
DRY up CI workflow and make it easier to run locally

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
Whenever I get to a new repository, one of the first things I check is the GitHub workflows files, since that gives me a nice hint of what I need to run locally to get my PRs green.

And that's what I did in this project, but since everything is docker based, it's a bit painful to copy-paste all those docker commands spread across multiple lines.

Until I found `bin/ci-test`!

So this change makes it so that CI configuration can reuse `bin/ci-test`, making `bin/ci-test` more discoverable I believe.

Also I maintained backwards compatibility with the previous way of running the script (build + run one suite), since that's useful too and I don't want to break anybody's workflow.

Note that building images with this script does not really work on M1, but that was also the case before. The same handling of `TARGET_PLATFORM` that `bin/docker-dev-shell` does needs to be added here. But I'll leave that for another time!

### CODE REVIEW

**Comment by deivid-rodriguez** [comment]
ℹ️ If this is accepted, then the required branches need to be changed in UI configuration since job names have changed.

**Review by jeffwidman — APPROVED** [review]
Makes sense to me!

Given what this touches, probably best to get a second review from someone else on the team who's been around a bit longer and may have more historical context for why it was originally wired up this way.

File: bin/ci-test
Comment (Nishnha) [inline bin/ci-test:70]: I think we would want to do `env.` since the variable is set as a workflow environment variable in https://github.com/dependabot/dependabot-core/pull/5525/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fL69

This would probably break if you tried to run `run_suite` locally, though. Maybe we could test if the `LOCAL_GITHUB_ACCESS_TOKEN` is set first and use `env.` if it isn't?

```suggestion
    --env "DEPENDABOT_TEST_ACCESS_TOKEN=${env.LOCAL_GITHUB_ACCESS_TOKEN}" \
```


see https://docs.github.com/en/actions/learn-github-actions/environment-variables

File: bin/ci-test
Comment (Nishnha) [inline bin/ci-test:2]: This usage is kind of difficult to read. I think we can leave more extensive comments

```suggestion
#/ usages: 
#/  - ci-test <SUITES>
#/  - ci-test build-core
#/  - ci-test build-core-ci
#/  - ci-test run-suite <SUITE>
```

File: bin/ci-test
Comment (Nishnha) [reply bin/ci-test:70]: > Do the environment variables get passed to the workflow run commands?

> and the variables are interpolated on the runner machine that runs your workflow

Oh yep they totally are, disregard!

**Review by Nishnha — APPROVED** [review]
I have a minor suggestion around documentation, but overall this looks like a nice improvement. 

Glad to see that the CI for this branch is indeed using the updated workflow file from this PR:
https://github.com/dependabot/dependabot-core/runs/7805915696?check_suite_focus=true#step:3:2

**Comment by deivid-rodriguez** [comment]
Thanks for having a look @Nishnha, your suggestion seems totally on point 👍!

This PR somewhat conflicts with https://github.com/dependabot/dependabot-core/pull/5536, so I guess we can wait until that PR is merged, and then decide what to do here.

**Pull request closed without merging**
